### DISCUSSION TITLE
SQS queue invoked from EventBridge? Docs?

### DISCUSSION THREAD

**Discussion by riyadshauk** [getlift/lift#171] [General]
How can I set up EventBridge to directly call an SQS consumer?

I originally tried to have EventBridge call a Lambda, which would send an SQS message. The problem with that approach was the SQS consumer wouldn't get triggered at all, after the original Lambda was called. If I were to invoke the Lambda via Postman; however, the SQS consumer would get triggered. (Maybe this was an IAM problem in the configuration between EventBridge --> Lambda --> SQS? Would EventBridge even need to know SQS exists in that case? anyways...)

If I manually hook up EventBridge directly to my SQS consumer, it works fine (and that might even be more efficient, anyways). But then the only problem with that solution is that there's manual configuration (within EventBridge) required.

Originally, I had automated all the configuration, with declarative code like this:
```typescript
import { handlerPath } from '@libs/handler-resolver';

export default {
  handler: `${handlerPath(__dirname)}/handler.main`,
  events: [
    {
      http: {
        method: 'post',
        path: '/crons/service/function',
      },
      eventBridge: {
        schedule: 'rate(3 hours)',
      },
    },
  ],
};
```
I just don't know where I'd put an `eventBridge` property in the SQS consumer configuration:
```typescript
import { handlerPath } from '@libs/handler-resolver';

export default {
  type: 'queue',
  worker: {
    handler: `${handlerPath(__dirname)}/handler.main`,
    timeout: 600,
  },
};
```

There may be a really simple way to do this that I (quite likely) just didn't notice, so I'd appreciate a pointer there.

If there isn't a way to automate this, then maybe a way that doesn't use Lift?

Thanks in advance!

UPDATE (just sharing my current permissions configuration in my `serverless.dev.ts` file, in case that helps shed light on things):
```typescript
    lift: {
      automaticPermissions: false,
    },
    provider: {
        // ... stuff omitted ...
      iamRoleStatements: [
        {
          Effect: 'Allow',
          Action: ['dynamodb:*'],
          Resource: '*',
        },
        {
          Effect: 'Allow',
          Action: ['sqs:*'],
          Resource: '*',
        },
        // to enable access to SQS from EventBridge
        // see: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-troubleshooting.html#eb-sqs-encrypted
        {
          Effect: 'Allow',
          Action: [
            'kms:Decrypt',
            'kms:GenerateDataKey',
          ],
          Resource: '*',
        },
      ],
  }
```

**Comment by mnapoli** [comment]
> I originally tried to have EventBridge call a Lambda, which would send an SQS message. The problem with that approach was the SQS consumer wouldn't get triggered at all, after the original Lambda was called.

It sounds to me like the best approach, unless I'm missing something. If I were you I would investigate why the SQS consumer wasn't triggered.
